src/sbuf/Stats.cc

```diff
@@ -61,38 +61,37 @@ SBufStats::operator +=(const SBufStats& ss)
     return *this;
 }
 
-std::ostream &
-SBufStats::dump(std::ostream& os) const
+void
+SBufStats::dump(std::ostream& yaml) const
 {
     MemBlobStats ststats = MemBlob::GetStats();
-    os <<
-       "SBuf stats:\nnumber of allocations: " << alloc <<
-       "\ncopy-allocations: " << allocCopy <<
-       "\ncopy-allocations from C String: " << allocFromCString <<
-       "\nlive references: " << live <<
-       "\nno-copy assignments: " << assignFast <<
-       "\nclearing operations: " << clear <<
-       "\nappend operations: " << append <<
-       "\nmove operations: " << moves <<
-       "\ndump-to-ostream: " << toStream <<
-       "\nset-char: " << setChar <<
-       "\nget-char: " << getChar <<
-       "\ncomparisons with data-scan: " << compareSlow <<
-       "\ncomparisons not requiring data-scan: " << compareFast <<
-       "\ncopy-out ops: " << copyOut <<
-       "\nraw access to memory: " << rawAccess <<
-       "\nNULL terminate C string: " << nulTerminate <<
-       "\nchop operations: " << chop <<
-       "\ntrim operations: " << trim <<
-       "\nfind: " << find <<
-       "\ncase-change ops: " << caseChange <<
-       "\nCOW completely avoided: " << cowAvoided <<
-       "\nCOW replaced with memmove(3): " << cowShift <<
-       "\nCOW requiring an empty buffer allocation: " << cowJustAlloc <<
-       "\nCOW requiring allocation and copying: " << cowAllocCopy <<
-       "\naverage store share factor: " <<
-       (ststats.live != 0 ? static_cast<float>(live)/ststats.live : 0) <<
-       std::endl;
-    return os;
+    const std::string indent("  ");
```

Comment:
Please do not use std::string in primary Squid code. In most cases (probably including this one), using SBuf is best. I am aware that this code reports SBuf stats.